### DISCUSSION TITLE
How would I simulate this kind of light absorption by gemstones?

### DISCUSSION THREAD

**Discussion by AbelToy** [google/filament#7145] [Q&A]
The designer for our app made this model in blender. It's just using the Principled BSDF shader with a custom gradient for the base color, which I could easily replicate with a custom material.

| ![badge](https://github.com/google/filament/assets/147130/5d315219-11a2-4f76-bee7-8053e88d1193) | ![animated badge](https://github.com/google/filament/assets/147130/b0104c77-eda7-4a82-8e8f-821ab520b9fd) |
--- | ---

However, when using the Blender Cycles renderer, a nice "light absorption" effect happens in the bottom part of the gemstone, where the light is scattered and takes in the color of the surface.

Setting up ior / transmission / roughness / etc. properties to match the blender settings doesn't replicate this effect. I assume this is because Cycles uses ray tracing, since the Eevee renderer also doesn't produce such results.

These are the blender settings:

<img width="176" alt="Blender settings" src="https://github.com/google/filament/assets/147130/afc77aef-cdd0-4f0d-9bb7-8bf88fd5adc4">

And this is how it currently looks when rendered using filament:

<img width="300" alt="Filament output" src="https://github.com/google/filament/assets/147130/de123b85-1f60-45b2-9da0-5ebddaf00b87">

I was wondering if there would be a way to replicate this kind of effect using Filament's PBR capabilities, or if such an effect could be simulated using a custom shader, and if it's the later case I would really appreciate it if someone could point me in the right direction, since my shader knowledge is very newbie.

Best regards!

**Comment by romainguy** [comment]
One of the keys to getting transmission/refraction to look good is to set a thickness. You are right however that the result will never be identical to a raytracer but you should be able to get something that looks similar enough. For instance here is the same object with different thickness and transmission parameters (there's no gradient, just a single color):

<img width="1136" alt="Screenshot 2023-09-06 at 8 36 10 AM" src="https://github.com/google/filament/assets/869684/a3ac574a-dbce-4119-a88c-6741606f0c41">
<img width="1136" alt="Screenshot 2023-09-06 at 8 36 24 AM" src="https://github.com/google/filament/assets/869684/9cd6d4a7-3df5-4d6e-83da-dbe0bf5955cc">
<img width="1136" alt="Screenshot 2023-09-06 at 8 36 26 AM" src="https://github.com/google/filament/assets/869684/6990560c-4824-42b8-a6ab-7a7e29c282fe">
<img width="1136" alt="Screenshot 2023-09-06 at 8 36 04 AM" src="https://github.com/google/filament/assets/869684/965dca38-7e6b-4b17-b222-46b04f1d4b41">

Another important thing to note is that Blender doesn't set roughness the same way as Filament (unless you export glTF from Blender). To go from Blender roughness to Filament roughness you need to take the square root of the roughness.
